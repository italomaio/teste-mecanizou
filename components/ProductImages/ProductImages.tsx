"use client";

import React, { useState } from "react";
import Image from "next/image";

export type ProductImageProps = {
  images: string[];
};

const ProductImages: React.FC<ProductImageProps> = ({ images }) => {
  const [currentImage, setCurrentImage] = useState<string>(images[0]);

  return (
    <div className="flex flex-row-reverse ">
      <div className="relative w-full aspect-[16/9]">
        <Image
          className="object-contain"
          src={currentImage}
          alt="Product Image"
          placeholder="empty"
          fill={true}
          priority={true}
          role="img"
          aria-label="Current product image"
        />
      </div>
      <div className="flex flex-col gap-2">
        {images.map((x, index) => (
          <Image
            aria-label={`View image ${index + 1}`}
            aria-pressed={currentImage === x}
            key={x}
            role="button"
            className="object-contain border border-gray-300 dark:border-zinc-700 p-2 rounded-sm cursor-pointer focus-visible-ring hover:dark:border-zinc-600 hover:dark:bg-zinc-700"
            src={x}
            alt="Product Image"
            width={70}
            height={70}
            loading="lazy"
            onClick={() => setCurrentImage(x)}
          />
        ))}
      </div>
    </div>
  );
};

export default ProductImages;
