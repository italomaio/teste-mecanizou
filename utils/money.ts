export const formatter = new Intl.NumberFormat("pt-BR", {
  style: "currency",
  currency: "BRL",
  minimumFractionDigits: 2,
});

export function calculateDiscountPrice(
  price: number,
  discountPercentage: number
) {
  const discountFactor = discountPercentage / 100;
  const newPrice = price * (1 - discountFactor);

  return {
    oldPrice: formatter.format(price),
    newPrice: formatter.format(newPrice),
  };
}
