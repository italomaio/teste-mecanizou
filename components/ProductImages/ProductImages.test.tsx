import React from "react";
import { render, screen, fireEvent } from "@/tests/utils";
import ProductImages from "./ProductImages";

describe("ProductImages", () => {
  const mockImages: string[] = ["/image1.jpg", "/image2.jpg", "/image3.jpg"];

  it("Should render correctly", () => {
    const { container } = render(<ProductImages images={mockImages} />);
    expect(container).toBeInTheDocument();
  });

  it("Should change image on click thumbnail", () => {
    render(<ProductImages images={mockImages} />);
    fireEvent.click(screen.getByLabelText(/View image 1/i));
    expect(screen.getByLabelText(/Current product image/i)).toHaveAttribute(
      "src",
      mockImages[0]
    );
  });

  it("Should match snapshot", () => {
    const { asFragment } = render(<ProductImages images={mockImages} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
