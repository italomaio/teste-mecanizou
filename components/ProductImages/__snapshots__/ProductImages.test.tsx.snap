// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ProductImages Should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex flex-row-reverse "
  >
    <div
      class="relative w-full aspect-[16/9]"
    >
      <img
        alt="Product Image"
        aria-label="Current product image"
        class="object-contain"
        role="img"
        src="/image1.jpg"
      />
    </div>
    <div
      class="flex flex-col gap-2"
    >
      <img
        alt="Product Image"
        aria-label="View image 1"
        aria-pressed="true"
        class="object-contain border border-gray-300 dark:border-zinc-700 p-2 rounded-sm cursor-pointer focus-visible-ring hover:dark:border-zinc-600 hover:dark:bg-zinc-700"
        height="70"
        role="button"
        src="/image1.jpg"
        width="70"
      />
      <img
        alt="Product Image"
        aria-label="View image 2"
        aria-pressed="false"
        class="object-contain border border-gray-300 dark:border-zinc-700 p-2 rounded-sm cursor-pointer focus-visible-ring hover:dark:border-zinc-600 hover:dark:bg-zinc-700"
        height="70"
        role="button"
        src="/image2.jpg"
        width="70"
      />
      <img
        alt="Product Image"
        aria-label="View image 3"
        aria-pressed="false"
        class="object-contain border border-gray-300 dark:border-zinc-700 p-2 rounded-sm cursor-pointer focus-visible-ring hover:dark:border-zinc-600 hover:dark:bg-zinc-700"
        height="70"
        role="button"
        src="/image3.jpg"
        width="70"
      />
    </div>
  </div>
</DocumentFragment>
`;
